core/src/main/java/org/apache/calcite/avatica/MetaImpl.java

```diff
@@ -388,6 +388,32 @@ public MetaColumn(
     public String getName() {
       return columnName;
     }
+    public static String[] getColumnNames() {
```

Comment:
Might as well link to some documentation: https://docs.oracle.com/en/java/javase/20/docs/api/java.sql/java/sql/DatabaseMetaData.html#getColumns(java.lang.String,java.lang.String,java.lang.String,java.lang.String)

Also add a comment explaining why anybody might want to override this.

Reply:
It's static so it can't be overridden, the reader thinks. The fact that the caller is using reflection to find and invoke it could definitely do with some commentary, if that is indeed the best way for the caller to get this information.